### PR TITLE
Play sound on shipwreck pickup

### LINKED ISSUE #1971: No sound playing, while shipwreck survivor was gathered
In the OG an ordinary "collecting sound" is playing, when the shipwreck survivor disappears.
In fheroes2 we can hear no sounds.
image

### PR DESCRIPTION
**Pull request by tau3**
Hi!

This one fixes #1971. There are also changes for inspections in corresponding funciton.

### CODE REVIEW

File: src/fheroes2/heroes/heroes_action.cpp [hunk 1]
```diff
@@ -1627,9 +1627,10 @@ void ActionToShipwreckSurvivor( Heroes & hero, u32 obj, s32 dst_index )
     Maps::Tiles & tile = world.GetTiles( dst_index );
 
     if ( hero.IsFullBagArtifacts() ) {
-        const u32 gold = GoldInsteadArtifact( obj );
+        const int object = static_cast<int>( obj );
+        const u32 gold = GoldInsteadArtifact( object );
         DialogWithGold(
-            MP2::StringObject( obj ),
+            MP2::StringObject( object ),
```
Comment (ihhub) [inline src/fheroes2/heroes/heroes_action.cpp:1633]: To solve the issue with type casting please change `u32 obj` input argument to `int obj`.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @tau3 I left only one comment. Could you please take a look?

**Comment by tau3** [comment]
Hi @ihhub! Yeah, that is obviously better. Done.

**Comment by ihhub** [comment]
@tau3 a bunch of thanks for fixing this issue!

**Pull request merged**